pgml-sdks/python/pgml/pgml/collection.py

```diff
@@ -0,0 +1,771 @@
+        """
+        conn = self.pool.getconn()
+        self.models_table = self.name + ".models"
+        create_statement = (
+            "CREATE TABLE IF NOT EXISTS %s (\
+                            id          serial8 PRIMARY KEY, \
+                            created_at  timestamptz NOT NULL DEFAULT now(), \
+                            task        text NOT NULL, \
+                            name        text NOT NULL, \
+                            parameters  jsonb NOT NULL DEFAULT '{}'\
+                    );"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, create_statement)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS created_at_index ON %s (created_at);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS task_index ON %s (task);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS name_index ON %s (name);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        index_statement = (
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS parameters_index ON %s USING GIN (parameters jsonb_path_ops);"
+            % (self.models_table)
+        )
+        run_create_or_insert_statement(conn, index_statement, autocommit=True)
+
+        self.pool.putconn(conn)
+
+    def _create_transforms_table(self) -> None:
+        """
+        This function creates a transforms table in a PostgreSQL database and adds indexes to it.
+        """
+        conn = self.pool.getconn()
+        self.transforms_table = self.name + ".transforms"
+        create_statement = (
+            "CREATE TABLE IF NOT EXISTS %s (\
+                            oid         regclass PRIMARY KEY,\
+                            created_at  timestamptz NOT NULL DEFAULT now(), \
+                            task        text NOT NULL, \
+                            splitter    int8 NOT NULL REFERENCES %s\
+                              ON DELETE CASCADE\
+                              ON UPDATE CASCADE\
+                              DEFERRABLE INITIALLY DEFERRED,\
+                            model       int8 NOT NULL REFERENCES %s\
+                              ON DELETE CASCADE\
+                              ON UPDATE CASCADE\
+                              DEFERRABLE INITIALLY DEFERRED,\
+                            UNIQUE (task, splitter, model)\
+                    );"
+            % (self.transforms_table, self.splitters_table, self.models_table)
+        )
+        run_create_or_insert_statement(conn, create_statement)
+
+        index_statement = (
```

Comment:
`UNIQUE (task, splitter, model)` in the table definition creates a compound index on those three columns. Having 3 more individual indexes on the same columns may not be necessary.